### PR TITLE
Fix image configuration

### LINKED ISSUE #1109: Helm Chart v1.10.0 broken image reference
The image is constructed as `docker.io/cr.k8ssandra.io/k8ssandra/k8ssandra-operator:v1.10.0` instead of `cr.k8ssandra.io/k8ssandra/k8ssandra-operator:v1.10.0`, which is a misconfiguration in the values.yaml.

### PR DESCRIPTION
**Pull request by pschichtel**
**What this PR does**:

Previously the image name was constructed as docker.io/cr.k8ssandra.io/... which is obviously wrong.

Fixes #1109

**Which issue(s) this PR fixes**:
I haven't created one, but I can if needed.

**Checklist**
- [X] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [X] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by adejanovski — CHANGES_REQUESTED** [review]
Thanks for pushing this PR promptly @pschichtel !

Would you mind adding an entry for this in the 1.10 changelog file before we merge it?

**Comment by pschichtel** [comment]
@adejanovski I amended the entry, is it fine like that? 1.10.0 is still considered unreleased in that file.

File: CHANGELOG/CHANGELOG-1.10.md [hunk 1]
```diff
@@ -26,3 +26,4 @@ When cutting a new release, update the `unreleased` heading to the tag being gen
 * [BUGFIX] [#1061](https://github.com/k8ssandra/k8ssandra-operator/issues/1061) Point to cass-config-builder 1.0.7 for arm64 compatibility
 * [ENHANCEMENT] [#956](https://github.com/k8ssandra/k8ssandra-operator/issues/956) Enable linting in the project
 * [BUGFIX] [#1102](https://github.com/k8ssandra/k8ssandra-operator/issues/1102) Update gRPC maximum receive size to 512MB. Note, the operator might need more max memory than the default to take advantage of this.
+* [BUGFIX] [#1109](https://github.com/k8ssandra/k8ssandra-operator/issues/1109) Correctly configure the cr.k8ssandra.io registry
```
Comment (adejanovski) [inline CHANGELOG/CHANGELOG-1.10.md:29]: **Issue:** This should go before line 18 in the "unreleased" section.

**Review by adejanovski — CHANGES_REQUESTED** [review]

File: CHANGELOG/CHANGELOG-1.10.md (hunk 1 above)
Comment (pschichtel) [reply CHANGELOG/CHANGELOG-1.10.md]: ops, I just realized how the file is structured.

**Review by adejanovski — APPROVED** [review]

**Comment by adejanovski** [comment]
Thanks @pschichtel, merged ✅

**Pull request merged**